llvm/lib/SYCLLowerIR/SYCLDeviceRequirements.cpp

```diff
@@ -0,0 +1,136 @@
+//===----- SYCLDeviceRequirements.cpp - collect data for used aspects ----=-==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/SYCLDeviceRequirements.h"
+
+#include "llvm/ADT/SmallString.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/IR/Module.h"
+#include "llvm/SYCLLowerIR/ModuleSplitter.h"
+#include "llvm/Support/PropertySetIO.h"
+
+#include <set>
+#include <vector>
+
+using namespace llvm;
+
+static int64_t ExtractSignedIntegerFromMDNodeOperand(const MDNode *N,
+                                                     unsigned OpNo) {
+  Constant *C = cast<ConstantAsMetadata>(N->getOperand(OpNo).get())->getValue();
+  return C->getUniqueInteger().getSExtValue();
+}
+static uint64_t ExtractUnsignedIntegerFromMDNodeOperand(const MDNode *N,
+                                                        unsigned OpNo) {
+  Constant *C = cast<ConstantAsMetadata>(N->getOperand(OpNo).get())->getValue();
+  return C->getUniqueInteger().getZExtValue();
+}
+static llvm::StringRef ExtractStringFromMDNodeOperand(const MDNode *N,
+                                                      unsigned OpNo) {
+  MDString *S = cast<llvm::MDString>(N->getOperand(OpNo).get());
+  return S->getString();
+}
+
+SYCLDeviceRequirements
+llvm::getSYCLDeviceRequirements(const module_split::ModuleDesc &MD) {
```

Comment:
Just a nit. Why do we need llvm:: prefix?
Also, based on how this function is called, I think it's more aptly called computeSYCLDeviceRequirements.

Thanks

Reply:
[Update getSYCLDeviceRequirements name](https://github.com/intel/llvm/pull/12727/commits/e4d6c2431e73d87461a0cdb234df474a8bccebdb)